### PR TITLE
Add pre-grant deviceId alternative (Rebased).

### PR DESCRIPTION
**Pull request by jan-ivar**
Rebase of https://github.com/w3c/mediacapture-main/pull/373. Issue https://github.com/w3c/mediacapture-main/issues/359.

### CODE REVIEW

**Comment by jan-ivar** [comment]
@martinthomson PTAL.

**Comment by stefhak** [comment]
lgtm.

File: getusermedia.html [hunk 1]
```diff
@@ -3067,29 +3067,33 @@
...
+              visibly inspect and delete the identifier, like a cookie.</p>
               <p class="fingerprint">Since <code>deviceId</code> may persist
               across browsing sessions and to reduce its potential as a
               fingerprinting mechanism, <code>deviceId</code> is to be treated
               as other persistent storage mechanisms such as cookies
-              [[COOKIES]]. User agents MUST reset per-origin device identifiers
-              when other persistent storages are cleared.</p>
+              [[COOKIES]], in that user agents MUST NOT persist device identifiers for
+              sites that are blocked from using cookies, and user agents MUST
+              reset per-origin device identifiers when other persistent storages
```
Comment (martinthomson) [inline getusermedia.html]: I think that storage is a singular always, so "storage is"

**Comment by alvestrand** [comment]
LGTM

**Pull request merged**